pmm/pmm2-dbaas-upgrade-tests.groovy

```diff
@@ -0,0 +1,370 @@
+library changelog: false, identifier: 'lib@master', retriever: modernSCM([
+    $class: 'GitSCMSource',
+    remote: 'https://github.com/Percona-Lab/jenkins-pipelines.git'
+]) _
+
+void runStagingServer(String DOCKER_VERSION, CLIENT_VERSION, VERSION_SERVICE_VERSION, ADMIN_PASSWORD) {
+    stagingJob = build job: 'aws-staging-start', parameters: [
+        string(name: 'DOCKER_VERSION', value: DOCKER_VERSION),
+        string(name: 'CLIENT_VERSION', value: CLIENT_VERSION),
+        string(name: 'VERSION_SERVICE_VERSION', value: VERSION_SERVICE_VERSION),
+        string(name: 'DOCKER_ENV_VARIABLE', value: ' -e ENABLE_DBAAS=1 -e PERCONA_TEST_VERSION_SERVICE_URL=https://check-dev.percona.com/versions/v1 -e PERCONA_TEST_DBAAS_PMM_CLIENT=perconalab/pmm-client:dev-latest'),
+        string(name: 'NOTIFY', value: 'false'),
+        string(name: 'DAYS', value: '1'),
+        string(name: 'ADMIN_PASSWORD', value: ADMIN_PASSWORD)
+    ]
+    env.VM_IP = stagingJob.buildVariables.IP
+    env.VM_NAME = stagingJob.buildVariables.VM_NAME
+    env.PMM_URL = "http://admin:${ADMIN_PASSWORD}@${VM_IP}"
+    env.PMM_UI_URL = "http://${VM_IP}/"
+}
+
+void runClusterStaging(String PMM_QA_GIT_BRANCH) {
+    clusterJob = build job: 'kubernetes-cluster-staging', parameters: [
+        string(name: 'NOTIFY', value: 'false'),
+        string(name: 'PMM_QA_GIT_BRANCH', value: PMM_QA_GIT_BRANCH),
+        string(name: 'DAYS', value: '1')
+    ]
+    env.CLUSTER_IP = clusterJob.buildVariables.IP
+    env.KUBECONFIG = clusterJob.buildVariables.KUBECONFIG
+}
+
+void destroyStaging(IP) {
+    build job: 'aws-staging-stop', parameters: [
+        string(name: 'VM', value: IP),
+    ]
+}
+
+def latestVersion = pmmVersion()
+def versionsList = pmmVersion('list')
+
+pipeline {
+    agent {
+        label 'agent-amd64'
+    }
+    parameters {
+        string(
+            defaultValue: 'main',
+            description: 'Tag/Branch for pmm-ui-tests repository',
+            name: 'GIT_BRANCH')
+        string(
+            defaultValue: '',
+            description: 'Commit hash for the branch',
+            name: 'GIT_COMMIT_HASH')
+        choice(
+            choices: ['2.34.0', '2.35.0'],
```

Comment:
I'd recommend use pmmVersion file we maintain, just add another for DbaaS versions https://github.com/Percona-Lab/jenkins-pipelines/blob/master/vars/pmmVersion.groovy